practice/src/js/templates/productTemplate.js

```diff
@@ -187,6 +188,25 @@ export default class ProductTemplate {
     Toast.error(MESSAGES.DELETE_PRODUCT_FAILED_MSG);
   }
 
+  /**
+   * Binds the search input to the search handler function and render the results.
+   * @param {Function} handlerSearchProduct - The handler function to fetch and return search results.
+   */
+  bindSearchProduct(handlerSearchProduct) {
+    const searchInput = document.getElementById('search-input');
+
+    if (!searchInput) {
+      return;
+    }
+
+    searchInput.addEventListener('keyup', async (e) => {
+      const name = e.target.value;
+      const searchedProducts = await handlerSearchProduct({ name });
```

Comment:
@JennyDoJD pls help refer to this link https://www.freecodecamp.org/news/javascript-debounce-example/. we need to apply `debounce` when typing to avoid spam API.